ste/jobStatusManager.go

```diff
@@ -41,6 +43,7 @@ type jobStatusManager struct {
 	listReq     chan bool
 	partCreated chan JobPartCreatedMsg
 	xferDone    chan xferDoneMsg
+	done        chan struct{}
```

Comment:
I see this channel is getting populatd from `CleanupJobStatusMgr` only once. My question is: why do we need it. Can we not print the "Cleanup done" at the end of `DeferredCleanupJobMgr`. I think you are already doing that. 
`jm.Log(pipeline.LogInfo, "DeferredCleanupJobMgr Exit, Closing the log")`
This way we will not need a channel and the thread performing the clean up can behave as a detached/independent goroutine (faster).

Reply:
done chan to tell handleStatusUpdateMessage goroutine to close.